api/v1alpha1/jwt_types.go

```diff
@@ -49,8 +53,17 @@ type JWTProvider struct {
 	Audiences []string `json:"audiences,omitempty"`
 
 	// RemoteJWKS defines how to fetch and cache JSON Web Key Sets (JWKS) from a remote
-	// HTTP/HTTPS endpoint.
-	RemoteJWKS RemoteJWKS `json:"remoteJWKS"`
+	// HTTP/HTTPS endpoint. This field is deprecated, https retrieval is supported by the uri
+	// type in JWKSSource
+	//
+	// +optional
+	RemoteJWKS *RemoteJWKS `json:"remoteJWKS"`
+
+	// JWKSSource defines how to fetch and cache JSON Web Key Sets (JWKS) from a given source.
+	// Inline, URI and Configmap/Secret contents are supported.
+	//
+	// +optional
+	JWKSSource *JWKSSource `json:"jwksSource"`
```

Comment:
I like the `jwks.Type` idea !
we have been meaning to also add a `backendRefs` field to `remoteJwks` so upstream connection/traffic parameters can be controlled, relates to https://github.com/envoyproxy/gateway/issues/3536


so the 2 options here are
1. Represent backendRefs as a `Type`
2. Keep the `remoteJwks` field, and add a new one `localJwks` to support the inline and valueRef cases

cc @envoyproxy/gateway-maintainers

Reply:
+1 for 2 as the structure is more straightfoward and less levels.


```
      remoteJWKS:
        backendRefs:
        - group: gateway.envoyproxy.io
          kind: Backend
          name: backend-fqdn
          port: 443
        backendSettings:
          retry:
            numRetries: 3
            perRetry:
              backOff:
                baseInterval: 1s
                maxInterval: 5s
            retryOn:
              triggers: ["5xx", "gateway-error", "reset"]
      localJWKS:
        type: inline
        vaule: xxxxxxx

```

```
    jwksSource:
      type: backendRefs
      backendRefs:
        backendRefs:
        - group: gateway.envoyproxy.io
          kind: Backend
          name: backend-fqdn
          port: 443
        backendSettings:
          retry:
            numRetries: 3
            perRetry:
              backOff:
                baseInterval: 1s
                maxInterval: 5s
            retryOn:
              triggers: ["5xx", "gateway-error", "reset"]

```

Reply:
+1 for 2

Reply:
thanks for sharing your thoughts folks, lets go with 2.